modules/api/php/endpoints/candidate/visit/image/format/downloadfile.class.inc

```diff
@@ -100,20 +102,47 @@ class Thumbnail extends Endpoint implements \LORIS\Middleware\ETagCalculator
      */
     private function _handleGET(): ResponseInterface
     {
-        $info = $this->_image->getThumbnailFileInfo();
+        switch ($this->_format) {
```

Comment:
I would put this `switch` in the constructor and assigne $this->file-type, $this->content_type ans $this->info (or filepointer) so that both _hanldeGET and Etag can access instance variable. That would prevent us from having to maintain 2 `switch` blocks.

Reply:
@xlecours Done! Thanks :)